src/sources/static_metrics.rs

```diff
@@ -0,0 +1,302 @@
+use std::collections::BTreeMap;
+use std::num::NonZero;
+use std::time::Duration;
+
+use chrono::Utc;
+use futures::StreamExt;
+use serde_with::serde_as;
+use tokio::time;
+use tokio_stream::wrappers::IntervalStream;
+use vector_lib::configurable::configurable_component;
+use vector_lib::internal_event::{CountByteSize, InternalEventHandle as _};
+use vector_lib::{config::LogNamespace, ByteSizeOf, EstimatedJsonEncodedSizeOf};
+
+use crate::{
+    config::{SourceConfig, SourceContext, SourceOutput},
+    event::{
+        metric::{MetricData, MetricName, MetricSeries, MetricTime, MetricValue},
+        EventMetadata, Metric, MetricKind,
+    },
+    internal_events::{EventsReceived, StaticMetricsBytesReceived, StreamClosedError},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+
+/// Configuration for the `static_metrics` source.
+#[serde_as]
+#[configurable_component(source(
+    "static_metrics",
+    "Produce static metrics defined in configuration."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct StaticMetricsConfig {
+    /// The interval between metric emitting, in seconds.
+    #[serde_as(as = "serde_with::DurationSecondsWithFrac<f64>")]
+    #[serde(default = "default_interval")]
+    #[configurable(metadata(docs::human_name = "Emitting interval"))]
+    pub interval_secs: Duration,
```

Comment:
I personally prefer to specify durations in milliseconds and use a u64 instead of f64. What do you think?

I dont know how is it usually done in other sinks/sources config, @jszwedko ?

Reply:
I have tried to keep it the same across sources in this case, so for this one I copied what was done in `internal_metrics`.

Reply:
Looking across other sources, seconds are used in general (either u64 or f64, depending on the source) and there are examples of milliseconds, but they are always explicitly specified, by adding the `_ms` suffix to the property name.